cmd/kpod/login.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"golang.org/x/crypto/ssh/terminal"
+
+	"github.com/containers/image/pkg/authentication"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+const defaultServer = "https://registry-1.docker.io"
+
+var (
+	loginFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "password, p",
+			Usage: "Password",
+		},
+		cli.StringFlag{
+			Name:  "username, u",
+			Usage: "Username",
+		},
+	}
+	loginDescription = "Login to a container registry on a specified server."
+	loginCommand     = cli.Command{
+		Name:        "login",
+		Usage:       "login to a container registry",
+		Description: loginDescription,
+		Flags:       loginFlags,
+		Action:      loginCmd,
+		ArgsUsage:   "SERVER",
+	}
+)
+
+// loginCmd uses the authentication package to store a user's authenticated credentials
+// in an auth.json file for future use
+func loginCmd(c *cli.Context) error {
+	args := c.Args()
+	if len(args) > 1 {
+		return errors.Errorf("too many arguments, login takes only 1 argument")
+	}
+	var server string
+	if len(args) == 1 {
+		server = args[0]
+	} else {
+		server = defaultServer
+	}
+
+	username, password, err := getUserAndPass(c.String("username"), c.String("password"))
+	if err != nil {
+		return errors.Wrapf(err, "error getting username and password")
+	}
+
+	if err = authentication.SetAuthentication(username, password, server); err != nil {
+		return errors.Wrapf(err, "error setting authentication")
+	}
+
+	fmt.Println("Login Succeeded!")
+	return nil
+}
+
+// getUserAndPass gets the username and password from STDIN if not given
+// using the -u and -p flags
+func getUserAndPass(username, password string) (string, string, error) {
+	var err error
+	reader := bufio.NewReader(os.Stdin)
+	if username == "" {
+		fmt.Print("Username: ")
+		username, err = reader.ReadString('\n')
+		if err != nil {
+			return "", "", errors.Wrapf(err, "error reading username")
+		}
+	}
+	if password == "" {
+		fmt.Print("Password: ")
+		pass, err := terminal.ReadPassword(0)
+		if err != nil {
+			return "", "", errors.Wrapf(err, "error reading password")
+		}
+		password = string(pass)
+		fmt.Println()
+	}
+	return strings.TrimSpace(username), strings.TrimSpace(password), err
```

Comment:
Do we not support passwords with trailing spaces?

Reply:
I think by convention, not technical reasons, most password systems don't support trailing spaces.  Speaking of format rules, is there one suggested/required by RedHat Security folks?

Reply:
I didn't think one could have a password with a trailing space. I can fix that to be supported.

Reply:
👎  to trailing space on pwd fwiw.

Reply:
Docker trims tabs and newlines but not spaces, so I think we should support trailing spaces (someone may have one in their password and be very confused as to why they can't log in)

Reply:
so what is the vote? trailing spaces in password or not?

Reply:
I would leave them.

Reply:
Are you using single quotes to enclose the password when setting it from the flag? That should preserve the whitespace

Reply:
@mheon I just realised that, sorry I was being silly.

Reply:
Happens to all of us :-)

Reply:
Fixed